lib/logitech_receiver/hidpp10.py

```diff
@@ -223,18 +223,23 @@ def get_firmware(device):
 
 	firmware = [None, None, None]
 
+	#workarounds for EX100, which doesnt support report firmware
```

Comment:
This chunk showd be executed after line 231 ( `if not reply:`) since it is only important if everything else failed, right?

Reply:
Hmm, yes, i think. Please take into account what I not a professional Python programmer. :) May be it's a rudiment of source hacking. Your variant are working too. If it important, i can fix this

Reply:
It is simply matter of organization not actually a bug. Regarding professionals? Well, Logitech is professional company and does not offer support for Linux. We (you, Daniel, Peter, Me, etc) do!